starlark/value.go

```diff
@@ -131,15 +131,36 @@ type Comparable interface {
 	CompareSameType(op syntax.Token, y Value, depth int) (bool, error)
 }
 
+// Alternate, simplified comparison interface
+type ThreeWayComparable interface {
+	Value
+	// ThreeWayCompareSameType compares one value to another of the same orderable
+	// Type() and returns a three-way comparison value (-1, 0, 1)
+	//
+	// Implementations that recursively compare subcomponents of
+	// the value should use the CompareDepth function, not Compare, to
+	// avoid infinite recursion on cyclic structures.
+	//
+	// The depth parameter is used to bound comparisons of cyclic
+	// data structures.  Implementations should decrement depth
+	// before calling CompareDepth and should return an error if depth
+	// < 1.
+	//
+	// Client code should not call this method.  Instead, use the
+	// standalone Compare or Equals functions, which are defined for
+	// all pairs of operands.
+	ThreeWayCompareSameType(y Value, depth int) (int, error)
+}
+
 var (
-	_ Comparable = Int{}
-	_ Comparable = False
-	_ Comparable = Float(0)
-	_ Comparable = String("")
-	_ Comparable = (*Dict)(nil)
-	_ Comparable = (*List)(nil)
-	_ Comparable = Tuple(nil)
-	_ Comparable = (*Set)(nil)
+	_ ThreeWayComparable = Int{}
```

Comment:
Floats are totally ordered too.

(So are Strings but the implementation of strings.Compare is obtusely inefficient.)